Author: Lennart Poettering <lennart@poettering.net>
Date:   Wed Nov 28 12:41:44 2018 +0100
Subject: mount: don't propagate errors from mount_setup_unit() further up
    
    If we can't process a specific line in /proc/self/mountinfo we should
    log about it (which we do), but this should not affect other lines, nor
    further processing of mount units. Let's keep these failures local.
    
    Fixes: #10874

Origin: upstream, https://github.com/systemd/systemd/commit/ba0d56f55f2073164799be714b5bd1aad94d059a
Bug: https://github.com/systemd/systemd/issues/10874
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/systemd/+bug/1755863

--- a/src/core/mount.c
+++ b/src/core/mount.c
@@ -1498,7 +1498,7 @@
 static int mount_load_proc_self_mountinfo(Manager *m, bool set_flags) {
         _cleanup_(mnt_free_tablep) struct libmnt_table *t = NULL;
         _cleanup_(mnt_free_iterp) struct libmnt_iter *i = NULL;
-        int r = 0;
+        int r;
 
         assert(m);
 
@@ -1514,7 +1514,6 @@
         if (r < 0)
                 return log_error_errno(r, "Failed to parse /proc/self/mountinfo: %m");
 
-        r = 0;
         for (;;) {
                 const char *device, *path, *options, *fstype;
                 _cleanup_free_ char *d = NULL, *p = NULL;
@@ -1543,12 +1542,10 @@
 
                 (void) device_found_node(m, d, true, DEVICE_FOUND_MOUNT, set_flags);
 
-                k = mount_setup_unit(m, d, p, options, fstype, set_flags);
-                if (r == 0 && k < 0)
-                        r = k;
+                (void) mount_setup_unit(m, d, p, options, fstype, set_flags);
         }
 
-        return r;
+        return 0;
 }
 
 static void mount_shutdown(Manager *m) {
